GameObjectExtensions.cs

```diff
@@ -0,0 +1,97 @@
+		{
+			var behaviour = Utils.CreateGameObjectWithBehaviour<TBehaviour>(parent);
+			behaviour.Initialize(args);
+			return behaviour;
+		}
+
+		public static TBehaviour CreateChild<TBehaviour>(this GameObject parent, string path, bool worldPositionStay = true)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour
+		{
+			var behaviour = Utils.InstantiateResource<TBehaviour>(path, parent, worldPositionStay);
+			behaviour.Initialize();
+			return behaviour;
+		}
+
+		public static TBehaviour CreateChild<TBehaviour, TArgs>(this GameObject parent, string path, TArgs args, bool worldPositionStay = true)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour<TArgs>
+		{
+			var behaviour = Utils.InstantiateResource<TBehaviour>(path, parent, worldPositionStay);
+			behaviour.Initialize(args);
+			return behaviour;
+		}
+
+		public static TBehaviour CreateChild<TBehaviour>(this GameObject parent, TBehaviour toClone)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour
+		{
+			var behaviour = Utils.CloneBehaviour(toClone, parent);
+			behaviour.Initialize();
+			return behaviour;
+		}
+
+		public static TBehaviour CreateChild<TBehaviour, TArgs>(this GameObject parent, TBehaviour toClone, TArgs args)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour<TArgs>
+		{
+			var behaviour = Utils.CloneBehaviour(toClone, parent);
+			behaviour.Initialize(args);
+			return behaviour;
+		}
+
+		public static TBehaviour ReplaceChild<TBehaviour>(this GameObject parent, MonoBehaviour toDestroy)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour
+		{
+			Object.Destroy(toDestroy.gameObject);
+			return parent.CreateChild<TBehaviour>();
+		}
+
+		public static TBehaviour ReplaceChild<TBehaviour, TArgs>(this GameObject parent, MonoBehaviour toDestroy, TArgs args)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour<TArgs>
+		{
+			Object.Destroy(toDestroy.gameObject);
+			return parent.CreateChild<TBehaviour, TArgs>(args);
+		}
+
+		public static TBehaviour ReplaceChild<TBehaviour>(this GameObject parent, MonoBehaviour toDestroy, string path)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour
+		{
+			Object.Destroy(toDestroy.gameObject);
+			return parent.CreateChild<TBehaviour>(path);
+		}
+
+		public static TBehaviour ReplaceChild<TBehaviour, TArgs>(this GameObject parent, MonoBehaviour toDestroy, string path, TArgs args)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour<TArgs>
+		{
+			Object.Destroy(toDestroy.gameObject);
+			return parent.CreateChild<TBehaviour, TArgs>(path, args);
+		}
+
+		public static TBehaviour ReplaceChild<TBehaviour>(this GameObject parent, MonoBehaviour toDestroy, TBehaviour toClone)
+			where TBehaviour : MonoBehaviour, IHierarchyBehaviour
+		{
+			Object.Destroy(toDestroy.gameObject);
+			return parent.CreateChild(toClone);
+		}
+
+		public static TBehaviour ReplaceChild<TBehaviour, TArgs>(this GameObject parent, MonoBehaviour toDestroy, TBehaviour toClone, TArgs args)
```

Comment:
@pmead As the `Replace` method is now called `ReplaceChild` do you think it should throw if you pass in a object to destroy if its not a child in the scene?

Reply:
I'd possibly go one step further and say it should also check that the parent (this gameobject) has the permission to replace or destroy the object by checking it is a child of this object.